sentry_sdk/integrations/litestar.py

```diff
@@ -0,0 +1,285 @@
+from typing import TYPE_CHECKING
+
+import sentry_sdk
+from sentry_sdk.consts import OP
+from sentry_sdk.integrations import DidNotEnable, Integration
+from sentry_sdk.integrations.asgi import SentryAsgiMiddleware
+from sentry_sdk.integrations.logging import ignore_logger
+from sentry_sdk.scope import Scope as SentryScope, should_send_default_pii
+from sentry_sdk.tracing import SOURCE_FOR_STYLE, TRANSACTION_SOURCE_ROUTE
+from sentry_sdk.utils import (
+    ensure_integration_enabled,
+    event_from_exception,
+    transaction_from_function,
+)
+
+try:
+    from litestar import Request, Litestar  # type: ignore
+    from litestar.handlers.base import BaseRouteHandler  # type: ignore
+    from litestar.middleware import DefineMiddleware  # type: ignore
+    from litestar.routes.http import HTTPRoute  # type: ignore
+    from litestar.data_extractors import ConnectionDataExtractor  # type: ignore
+    from pydantic import BaseModel  # type: ignore
+
+    if TYPE_CHECKING:
+        from typing import Any, Dict, List, Optional, Union
+
+        from litestar.types.asgi_types import ASGIApp  # type: ignore
+        from litestar.types import (  # type: ignore
+            HTTPReceiveMessage,
+            HTTPScope,
+            Message,
+            Middleware,
+            Receive,
+            Scope as LitestarScope,
+            Send,
+            WebSocketReceiveMessage,
+        )
+        from litestar.middleware import MiddlewareProtocol
+        from sentry_sdk._types import Event, Hint
+except ImportError:
+    raise DidNotEnable("Litestar is not installed")
+
+
+_DEFAULT_TRANSACTION_NAME = "generic Litestar request"
+
+
+class LitestarIntegration(Integration):
+    identifier = "litestar"
+    origin = f"auto.http.{identifier}"
+
+    @staticmethod
+    def setup_once() -> None:
+        patch_app_init()
+        patch_middlewares()
+        patch_http_route_handle()
+
+        # logs an error for every 500
```

Comment:
This is intended to follow the pattern found in other integrations such as `DjangoIntegration.setup_once`.

Litestar `ExceptionHandlerMiddleware.__call__` catches exceptions and does the following (among other things):
1. Logs them, some at least as errors (such as 500s)
2. Calls `after_exception` hooks

The `LitestarIntegration` (like the `StarliteIntegration` before it) provides an `after_exception` hook to create a Sentry event from an exception, which ends up being called during step 2 above. However, the LoggingIntegration will by default create a Sentry event from error logs made in step 1 if we do not prevent it from doing so.

The existing `StarliteIntegration` does not require a similar line, but that is only because for its equivalent to step 1 above it only ever logs to debug (not error). If manually changed to log to error instead, it has the same issue that motivated adding this line.